evil_mystery_word.py

```diff
@@ -0,0 +1,112 @@
+from mystery_word import display_text, handle_input, get_difficulty, \
+    play_again, guess_checker, win_or_lose, final_message
+from collections import defaultdict
+import random
+import os
+
+
+# Functions
+
+# Get word length.
+def get_word_length(difficulty):
+    if difficulty == 1:
+        return random.randint(4, 6)
+    elif difficulty == 2:
+        return random.randint(6, 10)
+    return random.randint(10, 15)
+
+
+# Get initial word list.
+def get_init_list(word_length):
+    with open("/usr/share/dict/words") as f:
+        word_list = f.readlines()
+    return [word[:-1].lower() for word in word_list
+            if len(word) - 1 == (word_length)]
+
+
+# Returns a dictionary of counts of letter found at index
+def letter_location_histogram(word_list, match_letter):
+    histogram = defaultdict(int)
+    for word in word_list:
+        for index, letter in enumerate(word):
+            if letter == match_letter:
+                histogram[index] += 1
+    return dict(histogram)
+
+
+# Takes a dictionary and returns a list of tuples of the top number of words.
+def top_items(dictionary, number):
+    pair_list = [pair for pair in dictionary.items()]
+    pair_list.sort(key=lambda pair: -pair[1])
+    return pair_list[:number]
+
+
+# Refine new list with guess
+def refine_list(word_list, letter):
+    matching_words = [word for word in word_list if letter in word]
+    non_matching_words = [word for word in word_list if letter not in word]
```

Comment:
there is a pretty cool way you can do the above two lines using sets. Look into it. :vertical_traffic_light: